aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,428 @@
+}
+
+func resourceAwsBatchComputeEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	log.Printf("[DEBUG] Update compute environment \"%s\".\n", computeEnvironmentName)
+
+	input := &batch.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(computeEnvironmentName),
+		ComputeResources:   &batch.ComputeResourceUpdate{},
+	}
+
+	if d.HasChange("service_role") {
+		log.Printf("[DEBUG] Service role of \"%s\" has change.\n", computeEnvironmentName)
+		input.ServiceRole = aws.String(d.Get("service_role").(string))
+	}
+	if d.HasChange("state") {
+		log.Printf("[DEBUG] State of \"%s\" has change.\n", computeEnvironmentName)
+		input.ServiceRole = aws.String(d.Get("state").(string))
+	}
+
+	if d.HasChange("compute_resources") {
+		log.Printf("[DEBUG] Compute resources of \"%s\" has change.\n", computeEnvironmentName)
+		computeResources := d.Get("compute_resources").([]interface{})
+		if len(computeResources) == 0 {
+			return fmt.Errorf("One compute environment is expected, but no compute environments are set")
+		}
+		computeResource := computeResources[0].(map[string]interface{})
+
+		if d.HasChange("compute_resources.0.desired_vcpus") {
+			log.Printf("[DEBUG] Desired vCpus of \"%s\" has change.\n", computeEnvironmentName)
+			input.ComputeResources.DesiredvCpus = aws.Int64(int64(computeResource["desired_vcpus"].(int)))
+		}
+		if d.HasChange("compute_resources.0.max_vcpus") {
+			log.Printf("[DEBUG] Max vCpus of \"%s\" has change.\n", computeEnvironmentName)
+			input.ComputeResources.MaxvCpus = aws.Int64(int64(computeResource["max_vcpus"].(int)))
+		}
+		if d.HasChange("compute_resources.0.min_vcpus") {
+			log.Printf("[DEBUG] Min vCpus of \"%s\" has change.\n", computeEnvironmentName)
+			input.ComputeResources.MinvCpus = aws.Int64(int64(computeResource["min_vcpus"].(int)))
+		}
+	}
+
+	if _, err := conn.UpdateComputeEnvironment(input); err != nil {
+		return err
+	}
+
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func waitComputeEnvironmentValid(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	for {
+		result, err := conn.DescribeComputeEnvironments(&batch.DescribeComputeEnvironmentsInput{
+			ComputeEnvironments: []*string{
+				aws.String(computeEnvironmentName),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(result.ComputeEnvironments) == 0 {
+			return fmt.Errorf("One compute environment is expected, but AWS return no compute environment")
+		} else if len(result.ComputeEnvironments) >= 2 {
+			return fmt.Errorf("One compute environment is expected, but AWS return too many compute environment")
+		}
+		computeEnvironment := result.ComputeEnvironments[0]
+		if *(computeEnvironment.Status) == "VALID" {
+			break
+		} else {
+			time.Sleep(1 * time.Second)
```

Comment:
Not sure 1 second is enough... perhaps doing it every 5 seconds would be better.
By the way, do you think it would be possible to use [such behaviour](https://github.com/terraform-providers/terraform-provider-aws/blob/b8d4e1570fc43a2acee6b6e47f63c9db6b067fa2/aws/resource_aws_alb.go#L192-L216) instead of making this function?

(same would go for the deletion)